cms/static/js/factories/program_editor.js

```diff
@@ -0,0 +1,9 @@
+define([
+    'js/programs/program_admin_app'
+],
+function(program_admin_app) {
+  'use strict';
+  return function(){
+    new program_admin_app();
```

Comment:
I don't understand this. It looks like program_admin_app is already a factory, as it just creates an instance of a different view.

```
(function() {
    'use strict';
    require([
            'js/programs/views/program_admin_app_view'
        ],
        function( ProgramAdminAppView ) {
            return new ProgramAdminAppView();
        }
    );
})();
```

It doesn't return a class, so this code can't create an instance like this. 

Am I misunderstanding this? It looks to me like you don't need this file at all, and you can just make program_admin_app be the listed factory file.